beep/featurize.py

```diff
@@ -128,44 +130,46 @@ def init_full_model(cls, processed_cycler_run, init_pred_cycle=10, mid_pred_cycl
         assert final_pred_cycle > mid_pred_cycle, 'Must have final_pred_cycle > mid_pred_cycle'
         ifinal = final_pred_cycle - 1  # python indexing
         imid = mid_pred_cycle - 1
-        iini = init_pred_cycle - 1
-        summary = processed_cycler_run.summary
-        assert len(processed_cycler_run.summary) > final_pred_cycle, 'cycle count must exceed final_pred_cycle'
+        assert len(processed_cycler_data['summary']['discharge_capacity']) > final_pred_cycle, 'cycle count must exceed final_pred_cycle'
         cycles_to_average_over = 40  # For nominal capacity, use median discharge capacity of first n cycles
 
-        interpolated_df = processed_cycler_run.cycles_interpolated
         X = pd.DataFrame(np.zeros((1, 20)))
         labels = []
         # Discharge capacity, cycle 2 = Q(n=2)
-        X[0] = summary.discharge_capacity[1]
+        X[0] = processed_cycler_data['summary']['discharge_capacity'][1]
         labels.append("discharge_capacity_cycle_2")
 
         # Max discharge capacity - discharge capacity, cycle 2 = max_n(Q(n)) - Q(n=2)
-        X[1] = max(summary.discharge_capacity[np.arange(final_pred_cycle)] - summary.discharge_capacity[1])
+        X[1] = max([x - processed_cycler_data['summary']['discharge_capacity'][1] for x in processed_cycler_data['summary']['discharge_capacity'][:final_pred_cycle]])
         labels.append("max_discharge_capacity_difference")
 
         # Discharge capacity, cycle 100 = Q(n=100)
-        X[2] = summary.discharge_capacity[ifinal]
+        X[2] = processed_cycler_data['summary']['discharge_capacity'][ifinal]
         labels.append("discharge_capacity_cycle_100")
 
         # Feature representing time-temperature integral over cycles 2 to 100
-        X[3] = np.nansum(summary.time_temperature_integrated[np.arange(final_pred_cycle)])
+        X[3] = np.nansum(processed_cycler_data['summary']['time_temperature_integrated'][:final_pred_cycle])
         labels.append("integrated_time_temperature_cycles_1:100")
 
         # Mean of charge times of first 5 cycles
-        X[4] = np.nanmean(summary.charge_duration[1:6])
+        X[4] = np.nanmean(processed_cycler_data['summary']['charge_duration'][1:6])
         labels.append("charge_time_cycles_1:5")
 
         # Descriptors based on capacity loss between cycles 10 and 100.
-        Qd_final = interpolated_df.discharge_capacity[interpolated_df.cycle_index == ifinal]
-        Qd_10 = interpolated_df.discharge_capacity[interpolated_df.cycle_index == 9]
+        _ixf = [i for i, x in enumerate(processed_cycler_data['cycles_interpolated']['cycle_index']) if x == ifinal]
+        Qd_final = np.array(processed_cycler_data['cycles_interpolated']['discharge_capacity'])[_ixf]
+        _ix9 = [i for i, x in enumerate(processed_cycler_data['cycles_interpolated']['cycle_index']) if x == 9]
+        Qd_10 = np.array(processed_cycler_data['cycles_interpolated']['discharge_capacity'])[_ix9]
 
```

Comment:
I'm a little confused what datatype _ixf and _ix9 are. They appear to be lists of integers, but then how do they work as indices in lines 168 and 170?